### PR TITLE
Upgrade shred

### PR DESCRIPTION
**Pull request by torkleyy**
This change is [Reviewable](https://reviewable.io/reviews/slide-rs/specs/346)

### CODE REVIEW

File: src/common.rs [hunk 1]
```diff
@@ -186,7 +186,7 @@ where
     );
 
     fn run(&mut self, (entities, errors, mut futures, mut pers): Self::SystemData) {
-        for (e, _) in (&*entities, &futures.check()).join() {
+        for (e, _) in (&*entities, &futures.mask().clone()).join() {
```
Comment (WaDelma) [inline src/common.rs]: This clone could potentially be removed if we had way to 'drain' storage while joining.

File: src/storage/data.rs [hunk 2]
```diff
@@ -114,18 +114,18 @@ impl<'a, T> SystemData<'a> for ReadStorage<'a, T>
 where
     T: Component,
 {
-    fn fetch(res: &'a Resources, id: usize) -> Self {
-        Storage::new(res.fetch(0), res.fetch(id))
+    fn fetch(res: &'a Resources) -> Self {
+        Storage::new(res.fetch(), res.fetch())
```
Comment (WaDelma) [inline src/storage/data.rs]: This is kinda unreadable if you don't know which types of resources these fetch (But the `reads` method does tell the types..).

File: src/common.rs (hunk 1 above)
Comment (torkleyy) [reply src/common.rs]: Don't we have that? I just fixed the warning here (`check`) is deprecated.

Comment (WaDelma) [reply src/common.rs]: Yeah actually we have it.

**Comment by kvark** [comment]
So the dynamic components get removed now?

**Comment by torkleyy** [comment]
Yes, see https://github.com/slide-rs/shred/pull/75

**Comment by torkleyy** [comment]
bors r+

cc @BartAdv

**Pull request merged**